File: backend/pawnscan/src/main/java/edu/cit/garbo/pawnscan/features/verification/VerificationService.java
package edu.cit.garbo.pawnscan.features.verification;

import edu.cit.garbo.pawnscan.features.verification.dto.SearchLogResponse;
import edu.cit.garbo.pawnscan.features.verification.dto.StolenMatchResponse;
import edu.cit.garbo.pawnscan.features.verification.dto.VerifySearchResponse;

import java.util.List;

public interface VerificationService {

    VerifySearchResponse verifySerial(String authenticatedEmail, String serial);

    SearchLogResponse logSearch(String authenticatedEmail, String serial);

    List<SearchLogResponse> getSearchHistory(String authenticatedEmail, int page, int size);

    List<StolenMatchResponse> getStolenMatches(String authenticatedEmail, int page, int size);
}

File: mobile/app/src/main/java/com/cit/pawnscan/features/business/api/VerificationService.kt
package com.cit.pawnscan.features.business.api

import retrofit2.Call
import retrofit2.http.GET
import retrofit2.http.Header
import retrofit2.http.Query

interface VerificationService {
    @GET("/api/verify/search")
    fun verifySerial(
        @Header("Authorization") authorization: String,
        @Query("serial") serial: String
    ): Call<VerifySearchResponse>

    @GET("/api/verify/history")
    fun getSearchHistory(
        @Header("Authorization") authorization: String,
        @Query("page") page: Int = 0,
        @Query("size") size: Int = 20
    ): Call<List<SearchLogResponse>>

    @GET("/api/verify/matches")
    fun getStolenMatches(
        @Header("Authorization") authorization: String,
        @Query("page") page: Int = 0,
        @Query("size") size: Int = 20
    ): Call<List<StolenMatchResponse>>
}
